parser/src/lexer.rs

```diff
@@ -300,7 +300,15 @@ where
                 value_text.push_str(&self.radix_run(10));
             }
 
-            let value = f64::from_str(&value_text).unwrap();
+            let value = match f64::from_str(&value_text) {
+                Ok(v) => v,
+                Err(_e) => {
+                    return Err(LexicalError {
+                        error: LexicalErrorType::OtherError("Invalid decimal literal".to_owned()),
+                        location: self.get_pos(),
+                    });
+                }
+            };
```

Comment:
```suggestion
            let value = match f64::from_str(&value_text).map_err(|e| LexicalError {
                        error: LexicalErrorType::OtherError("Invalid decimal literal".to_owned()),
                        location: self.get_pos(),
                    })?;
```

Reply:
Thank you , `map_err` is better, i should remember this way.

Reply:
The `e` in closure must be ignored to prevent compiling warning, so i don't commit  your suggestion directly.

Reply:
yep, I suggest on the web but you write code in a full dev environment. don't worry about it. I usually trust your additional revision than my suggestion.